virtualizarr/kerchunk.py

```diff
@@ -61,10 +62,26 @@ def _automatically_determine_filetype(filepath: str) -> str:
     file_extension = Path(filepath).suffix
 
     if file_extension == ".nc":
-        # TODO how can we automatically distinguish netCDF3 and 4?
-        raise NotImplementedError(
-            "Cannot unambiguously automatically determine which kerchunk file format reader to use"
-        )
+        # checks if netCDF library is installed.
+        # It currently is not a requirement in the pyproj.toml.
+
+        if importlib.util.find_spec("netCDF4") is None:
+            raise ImportError(
+                "netCDF4 library is required to determine NetCDF file type."
+            )
+
+        import netCDF4
+
+        with netCDF4.Dataset(filepath, "r") as dataset:
+            if dataset.data_model == "NETCDF4":
+                filetype = "netCDF4"
+            elif dataset.data_model == "NETCDF3_CLASSIC":
+                filetype = "netCDF3"
+            else:
+                raise NotImplementedError(
```

Comment:
This is cool! Presumably this has to open the file to read this `data_model` information? Is that likely to incur any significant cost over the opening that `kerchunk` already needs to do?

Reply:
I think the read done by the NetCDF4 library is only reading attributes and not loading the entire file. Not 100% sure on this though.

Reply:
You can do this without opening the file with netCDF at all. Both file types have a "magic" at the beginning of the file.

```python
def guess_file_type(fp) -> FileType:
    magic = fp.read(4)
    fp.seek(0)
    if magic[:3] == b"CDF":
        return FileType.netcdf3
    elif magic == b"\x89HDF":
        return FileType.hdf5
    else:
        raise ValueError(f"Unknown file type - magic {magic}")
```